wagtail/wagtailadmin/edit_handlers.py

```diff
@@ -25,26 +25,43 @@
 from wagtail.utils.deprecation import RemovedInWagtail11Warning
 
 
+# Form field properties to override whenever we encounter a model field
+# that matches one of these types - including subclasses
 FORM_FIELD_OVERRIDES = {
     models.DateField: {'widget': widgets.AdminDateInput},
     models.TimeField: {'widget': widgets.AdminTimeInput},
     models.DateTimeField: {'widget': widgets.AdminDateTimeInput},
-    TaggableManager: {'widget': widgets.AdminTagWidget}
+    TaggableManager: {'widget': widgets.AdminTagWidget},
+}
+
+# Form field properties to override whenever we encounter a model field
+# that matches one of these types exactly, ignoring subclasses.
+# (This allows us to override the widget for models.TextField, but leave
+# the RichTextField widget alone)
+DIRECT_FORM_FIELD_OVERRIDES = {
```

Comment:
What if I wanted to make a subclass of TextField? It wouldn't be obvious to me that I would have to override the widget on that too.

Why not something like a ``wagtail_formfield`` method on the field? This would make ``FORM_FIELD_OVERRIDES`` just a thing that changes the Django default but the field still has the power to override Wagtail.